cpp/tests/sg/rf_test.cu

```diff
@@ -181,28 +186,70 @@ std::ostream& operator<<(std::ostream& os, const RfTestParams& ps)
+  } else {
+    // For regressors, no need to post-process predictions from FIL
+    pred = std::make_shared<thrust::device_vector<LabelT>>(params.n_rows * params.n_labels);
+    workspace = pred;
+  }
   TreeliteModelHandle model;
-  std::size_t num_outputs = 1;
-  if constexpr (std::is_integral_v<LabelT>) { num_outputs = params.n_labels; }
   build_treelite_forest(&model, forest, params.n_cols);
-  fil::treelite_params_t tl_params{fil::algo_t::ALGO_AUTO,
-                                   num_outputs > 1,
-                                   1.f / num_outputs,
-                                   fil::storage_type_t::AUTO,
-                                   8,
-                                   1,
-                                   0,
-                                   nullptr};
-  fil::forest_variant forest_variant;
-  fil::from_treelite(handle, &forest_variant, model, &tl_params);
-  fil::forest_t<float> fil_forest = std::get<fil::forest_t<float>>(forest_variant);
-  fil::predict(handle, fil_forest, pred->data().get(), X_transpose, params.n_rows, false);
+
+  auto filex_model = ML::experimental::fil::import_from_treelite_handle(
+    model,
+    ML::experimental::fil::tree_layout::breadth_first,
+    128,
+    false,
+    raft_proto::device_type::gpu,
+    handle.get_device(),
+    handle.get_stream());
+  filex_model.predict(handle,
+                      workspace->data().get(),
+                      X_transpose,
+                      params.n_rows,
+                      raft_proto::device_type::gpu,
+                      raft_proto::device_type::gpu,
+                      ML::experimental::fil::infer_kind::default_kind,
+                      1);
+  handle.sync_stream();
+
+  if constexpr (std::is_integral_v<LabelT>) {
+    // Perform argmax to convert probabilities into class outputs
+    auto offsets_it = thrust::make_transform_iterator(
+      thrust::make_counting_iterator(0),
+      [=] __device__ (int x) -> int { return x * params.n_labels; });
+    using kv_type = cub::KeyValuePair<int, DataT>;
+
+    // Compute size of workspace for the segmented reduce operation
+    std::size_t temp_storage_bytes = 0;
+    cub::DeviceSegmentedReduce::ArgMax(
+      nullptr, temp_storage_bytes, workspace->begin(),
+      thrust::device_pointer_cast<kv_type>(nullptr),
+      params.n_rows, offsets_it, offsets_it + 1);
+
+    // Allocate workspace and perform segmented reduce
+    thrust::device_vector<kv_type> workspace2(
+      params.n_rows + temp_storage_bytes / sizeof(kv_type) + 1);
+    cub::DeviceSegmentedReduce::ArgMax(
+      thrust::raw_pointer_cast(workspace2.data() + params.n_rows), temp_storage_bytes,
+      workspace->begin(), workspace2.begin(),
+      params.n_rows, offsets_it, offsets_it + 1);
+    thrust::transform(workspace2.begin(), workspace2.begin() + params.n_rows, pred->begin(),
+      [] __device__ (kv_type x) -> int { return x.key; });
```

Comment:
Note. The new FIL always returns probabilities for classifiers. If class outputs are desired, we need to perform argmax.